### PR TITLE
#166374539 Implement Favourites model

### PR DESCRIPTION
**Pull request by henryade**
#### What does this PR do?
Implement Favourites model

#### Description of Task to be completed?
Adds favourites model
adds associations with the user and author model 

#### How should this be manually tested?
- clone the repository and checkout to this branch
- run `npm install` to add new dependencies
- update/create the env file with your database details using the development node_env
- run `npm run db:roll` command
- check your database to see if tables are created
- you can run `npm run db:back` to remove tables 

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#166374539](https://www.pivotaltracker.com/story/show/166374539)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

**Comment by JuwonAbiola** [comment]
please check your commit message

**Comment by henryade** [comment]
> please check your commit message

Done

**Review by JuwonAbiola — DISMISSED** [review]
nice work

File: server/database/models/favourites.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+'use strict';
+module.exports = (sequelize, DataTypes) => {
+  const Favourites = sequelize.define('Favourites', {
+    id: DataTypes.INTEGER,
+    userId: DataTypes.INTEGER,
+    authorId: DataTypes.INTEGER
+  }, {});
+  Favourites.associate = function(models) {
+    // associations can be defined here
+    Favourites.hasOne(models.Users, {
```
Comment (Ukhu) [inline server/database/models/favourites.js]: Use 'belongsTo' instead

File: server/database/models/favourites.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      foreignKey: 'userId',
+      onDelete: 'CASCADE'
+    });
+    Favourites.hasOne(models.Author, {
```
Comment (Ukhu) [inline server/database/models/favourites.js]: Use 'belongsTo' instead

File: server/database/models/users.js [hunk 3]
```diff
@@ -11,6 +11,10 @@ module.exports = (sequelize, DataTypes) => {
   }, {});
   Users.associate = function(models) {
     // associations can be defined here
+    Users.hasMany(models.Favourites, {
+      foreignKey: 'authorId',
```
Comment (Ukhu) [inline server/database/models/users.js]: Use 'userId' here.

Comment (henryade) [reply server/database/models/users.js]: done

**Review by Ukhu — APPROVED** [review]

**Review by JuwonAbiola — APPROVED** [review]

**Review by allebd — APPROVED** [review]
Good one

**Review by KvNGCzA — APPROVED** [review]

**Pull request merged**